File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.py
class Solution:
  def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
    for i in range(len(points)):
      points[i] = (points[i][0]**2 + points[i][1]**2, points[i])
    heapify(points)
    res = []
    for i in range(k):
      res.append(heappop(points)[1])
    return res

File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.ts
function kClosest(points: number[][], k: number): number[][] {
  return points.map(([x, y]) => [x, y, x ** 2 + y ** 2])
         .sort(([,,a], [,,b]) => a - b)
         .slice(0, k)
         .map(([x, y]) => [x, y]);
};